### PR TITLE
CHDEXCHAE-64 / Search field dataset to reference comments text

### PR DESCRIPTION
**Pull request by mutantsan**

### CODE REVIEW

File: ckanext/data_qld/plugin.py [hunk 1]
```diff
@@ -2,15 +2,18 @@
 
 import logging
 
-from ckan import plugins
-
-from . import actions, auth_functions as auth, blueprints, click_cli, \
-    constants, converters, datarequest_auth_functions, helpers, validation
 import ckantoolkit as tk
 
+from ckan import model
```
Comment (ThrawnCA) [inline ckanext/data_qld/plugin.py:7]: These two lines could be combined: `from ckan import model, plugins`

File: ckanext/data_qld/templates/package/resource_read.html [hunk 2]
```diff
@@ -110,11 +110,20 @@ <h2>{{ _('Data Dictionary') }}</h2>
...
+                <i class="fa fa-lock"></i>
+                {{ _("Hidden" )}}
+            </span>
+        {% endif %}
+
+        {{ h.resource_display_name(res)}}
+
+        {% if res.schema %}
+            {{ h.get_validation_badge(res)|safe }}
+            {% set type = 'asset' if h.is_ckan_29() else 'resource' %}
```
Comment (ThrawnCA) [inline ckanext/data_qld/templates/package/resource_read.html:125]: CKAN 2.8 support has been dropped, this helper is going away.

File: config/project.ini
Comment (ThrawnCA) [inline config/project.ini:1]: We already have top-level `test.ini`, `.docker/test.ini`, and `.docker/ckan.ini`. Surely we don't need a fourth one?

**Review by ThrawnCA — CHANGES_REQUESTED** [review]

File: config/project.ini
Comment (mutantsan) [reply config/project.ini:1]: We need it for our workflow. This file shouldn't interfere with your setup. Could you delete it after we are done with our collaboration"?

File: ckanext/data_qld/templates/package/resource_read.html (hunk 2 above)
Comment (mutantsan) [reply ckanext/data_qld/templates/package/resource_read.html]: Fixed. I've preserved the style from your repo.

File: ckanext/data_qld/plugin.py (hunk 1 above)
Comment (mutantsan) [reply ckanext/data_qld/plugin.py]: Fixed

File: ckanext/data_qld/listeners.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import logging
```
Comment (duttonw) [inline ckanext/data_qld/listeners.py:1]: why could this not live inside ytp comments?

File: ckanext/data_qld/plugin.py [hunk 4]
```diff
@@ -199,6 +201,38 @@ def before_update(self, context, current_resource, updated_resource):
     def before_show(self, resource_dict):
         resource_freshness_helpers.process_nature_of_change(resource_dict)
 
+    def before_index(self, pkg_dict):
```
Comment (duttonw) [inline ckanext/data_qld/plugin.py:204]: why could this not live in ytp plugin to keep it self contained?

**Comment by duttonw** [comment]
@mutantsan , seperations of concerns, can you get ytp extension to be self contained with your signals upgrade. Please note that it should not enable signals if not on 2.10+ (does not exist in 2.9). See https://github.com/qld-gov-au/ckanext-ytp-comments/pull/104 where I've pull in the ytp only parts from this ticket.

**Pull request merged**